src/Microsoft.Azure.CosmosRepository/Repositories/DefaultRepository.cs

```diff
@@ -1,97 +1,98 @@
-            IRepositoryExpressionProvider repositoryExpressionProvider,
-            ISpecificationEvaluator specificationEvaluator)
-        {
-            _optionsMonitor = optionsMonitor;
-            _containerProvider = containerProvider;
-            _logger = logger;
-            _cosmosQueryableProcessor = cosmosQueryableProcessor;
-            _repositoryExpressionProvider = repositoryExpressionProvider;
-            _specificationEvaluator = specificationEvaluator;
-        }
-
-        static void TryLogDebugDetails(ILogger logger, Func<string> getMessage)
-        {
-            // ReSharper disable once ConstantConditionalAccessQualifier
-            if (logger?.IsEnabled(LogLevel.Debug) ?? false)
-            {
-                // ReSharper disable once TemplateIsNotCompileTimeConstantProblem
-                logger.LogDebug(getMessage());
-            }
-        }
-
-        static async Task<(List<TItem> items, double charge, string? continuationToken)> GetAllItemsAsync(
-            IQueryable<TItem> query,
-            int pageSize,
-            CancellationToken cancellationToken = default)
-        {
-            string? continuationToken = null;
-            List<TItem> results = new();
-            int readItemsCount = 0;
-            double charge = 0;
-            using FeedIterator<TItem> iterator = query.ToFeedIterator();
-            
-            while (readItemsCount < pageSize && iterator.HasMoreResults)
-            {
-                FeedResponse<TItem> next =
-                    await iterator.ReadNextAsync(cancellationToken)
-                        .ConfigureAwait(false);
-
-                foreach (TItem result in next)
-                {
-                    if (readItemsCount == pageSize)
-                    {
-                        break;
-                    }
-
-                    results.Add(result);
-                    readItemsCount++;
-                }
-
-                charge += next.RequestCharge;
-                continuationToken = next.ContinuationToken;
-            }
-
-            return (results, charge, continuationToken);
-        }
-    }
+﻿// Copyright (c) IEvangelist. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Azure.Cosmos;
+using Microsoft.Azure.Cosmos.Linq;
+using Microsoft.Azure.CosmosRepository.Options;
+using Microsoft.Azure.CosmosRepository.Processors;
+using Microsoft.Azure.CosmosRepository.Providers;
+using Microsoft.Azure.CosmosRepository.Specification.Evaluator;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+
+// ReSharper disable once CheckNamespace
+namespace Microsoft.Azure.CosmosRepository
+{
+    /// <inheritdoc/>
+    internal sealed partial class DefaultRepository<TItem> : IRepository<TItem>
+        where TItem : class, IItem
```

Comment:
I believe that we should be able to drop the `class` constraint here. Also, we only technically need one implementation of the `partial` to have the generic type constraints. That is why I only had it on this core implementation here (I'm ok with having it on both though, if you prefer that).

Reply:
Yeah I did think that, I think I did a bad merge. Updated.